src/NBitcoin/ConcurrentChain.cs

```diff
@@ -24,17 +24,17 @@ public ConcurrentChain()
         {
         }
 
-        public ConcurrentChain(BlockHeader genesis)
+        public ConcurrentChain(BlockHeader genesisHeader)
         {
-            this.SetTip(new ChainedBlock(genesis, 0));
+            this.SetTip(new ChainedBlock(genesisHeader, genesisHeader.GetHash(), 0));
```

Comment:
I am not sure you can just change the overload they do different things.

Reply:
The effect should be identical as "headerHash" replaces the "header.GetHash()" in the original method. Here the GetHash is more accessible to shallower call levels for passing in the network options - which still has to happen. Elsewhere, where the NetworkOptions are available, we are passing them in to similar calls as this one.

Reply:
Ok if behaviour is the same then no problem